#43 Feed Aggregator

lrowang opened the pull request:
PR for #43 

Hi @lhridley -

Pushing these commits for your WIP review. I have a sneaking suspicion that somehow my feature is not quite capturing all of the Aggregator Feed's configurations. Do I have to write a custom hook_update the captures the feed just like `forums` and `about us` pages?

Here's what I did:
1. Created the "Latest AS News" feed using the Aggregator module
2. Configured the newly created feed's Block so that it shows up on the Homepage, under Sidebar2 (Pic1)
3. Created a feature module that captures the block configuration `block_feeds_latest_news`
(Pic2)
4. Enabled the feature via hooks

- **This seems right to me, but it also feels like I might be missing a step 1.5 where I need to save the Aggregator module's configuration. I can see where the new "item" got created in the `aggregator_feed` table in the DB -- is `block_feeds_latest_news` capturing it? (pic3)**

- Lastly, would it be better to use the "Feeds" module instead of "Aggregator"?

### Pic1
![cs 43-home](https://cloud.githubusercontent.com/assets/19154706/19502527/5533d940-9562-11e6-8e25-39e28e5571a1.png)

### Pic2
![cs 43-2](https://cloud.githubusercontent.com/assets/19154706/19502532/66474eb0-9562-11e6-9cf8-18f24fa1012d.png)

### Pic3
![cs -3](https://cloud.githubusercontent.com/assets/19154706/19502535/6e104a20-9562-11e6-9e4e-c3982f7f4517.png)

lhridley commented:
In looking at the Aggregator module and doing a little googling, it appears that the Aggregator module does not expose its data as exportable, which is a requirement before it can be exposed to Features (https://www.drupal.org/node/2363833).  So, you will need to write a hook update / hook install method to actually get your "Latest News" aggregator into code:

http://drupal.stackexchange.com/questions/52800/how-to-export-feed-agregator-list-and-setting-as-feature/120077#120077

lrowang commented:
Hi @lhridley -

PR is ready for official review!
- Added the Aggregator feed via hook update/install (thank you for providing the above resource!).
- Added the cron call and clear cache call to the 'make provision` target
- Updated the permissions so all roles can view the Feed and adjusted test suite to match
- Created new test suite to check the Feed

I did not re-create the feed using the "Feeds" module because I ran out of time, but I made a note of it to try as optional/nice-to-have at the end of the project.

One last **question:**
If the "Latest As News" feed gets created via a hook_update instead of hook_install, when would the cron job call get an opportunity to be executed? Do I need to include it in the same hook_update?
